policy/predicate/signature.go

```diff
@@ -27,26 +27,40 @@ type HasValidSignatures bool
 
 var _ Predicate = HasValidSignatures(false)
 
-func (pred HasValidSignatures) Evaluate(ctx context.Context, prctx pull.Context) (bool, string, error) {
+func (pred HasValidSignatures) Evaluate(ctx context.Context, prctx pull.Context) (*common.PredicateResult, error) {
 	commits, err := prctx.Commits()
+
+	predicateResult := common.PredicateResult{
+		Satisfied:       false,
+		ConditionPhrase: "have valid signatures",
+	}
+
 	if err != nil {
-		return false, "", errors.Wrap(err, "failed to get commits")
+		return &predicateResult, errors.Wrap(err, "failed to get commits")
 	}
 
+	var commitHashes []string
+
 	for _, c := range commits {
 		valid, desc := hasValidSignature(ctx, c)
+		commitHashes = append(commitHashes, c.SHA)
 		if !valid {
+			predicateResult.Values = []string{c.SHA}
 			if pred {
-				return false, desc, nil
+				predicateResult.Description = desc
+				return &predicateResult, nil
```

Comment:
These predicates look like another place where `Satisfied` was set to false upfront instead of right before returning.